commons/client/discovery/discovery_client.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+"""
+DiscoveryClient is responsible for providing a list of service's instances.
+There may be many services, for example: 'user-service', 'order-service', 'product-service', and so on.
+Where each service may have several instances registered (e.g. user-service-1, user-service-2, ...),
+and the registered instances are called ServiceInstances.
+"""
+
+# standard library
+from abc import ABC, abstractmethod
+from typing import List, Set
+
+# scip plugin
+from commons.client.service_instance import ServiceInstance, StaticServiceInstance
+from commons.utils.functional_operators import filter_get_first
+
+__author__ = "Waterball (johnny850807@gmail.com)"
+__license__ = "Apache 2.0"
+
+
+class DiscoveryClient(ABC):
+    @abstractmethod
+    def get_instances(self, service_id: str) -> List[ServiceInstance]:
+        """
+        Gets all ServiceInstances associated with a particular serviceId.
+        :param service_id: (str) The serviceId to query.
+        :return: A List of ServiceInstance.
+        """
+        pass
+
+    @property
+    @abstractmethod
+    def services(self) -> List[str]:
+        """
+        :return: All known service IDs (*str).
+        """
+        pass
+
+
+class StaticDiscoveryClient(DiscoveryClient):
```

Comment:
@TsengMJ `DiscoveryClient` is an interface, while `StaticDiscoveryClient` is one of the implementations